drakrun/drakrun/lib/postprocessing/capa_plugin/capa_processor.py

```diff
@@ -0,0 +1,452 @@
+    elif isinstance(address, capa.features.address.ThreadAddress):
+        return {**format_capa_address(address.process), "tid": address.tid}
+    elif isinstance(address, capa.features.address.DynamicCallAddress):
+        return {**format_capa_address(address.thread), "cid": address.id}
+    elif isinstance(address, capa.features.address._NoAddress):
+        return {"address": "no address"}
+    else:
+        logger.debug("Encountered unknown address type: %s", type(address))
+        return {"address": address}
+
+
+def construct_ttp_block(
+    rule: capa.rules.Rule, addresses: capa.features.address
+) -> Dict[str, Any]:
+    return {
+        "name": rule.name.split("/")[0],
+        "mbc": rule.meta.get("mbc", None),
+        "att&ck": rule.meta.get("att&ck", None),
+        "occurances": [
+            format_capa_address(address=address) for address, _ in addresses
+        ],
+    }
+
+
+def construct_ttp_blocks(
+    rules: capa.rules.RuleSet,
+    capabilities_per_file: List[Tuple[pathlib.Path, capa.engine.MatchResults, Any]],
+    filter_function=lambda rule: rule
+) -> Iterator[Dict[str, Any]]:
+
+    # construct a ttp block for each extracted capability
+    for _, capabilities in capabilities_per_file:
+        for name, addresses in capabilities.items():
+            if filter_function(rules[name]):
+                yield construct_ttp_block(rules[name], addresses)
+
+
+def capa_analysis(analysis_dir: pathlib.Path) -> None:
+    # check and prepare the rules folder
+    if not check_rules_exist(capa_rules_dir):
+        # in case of a missing/empty rules folder, clone the official capa rules
+        clone_rules_reporsitory(capa_rules_dir)
+
+    # get rules and filter them
+    rules = get_rules([capa_rules_dir])
+    rules = filter_rules(
+        rules,
+        filter_function=lambda rule: rule.meta.get("att&ck", None)
+    )  # select only rules with an att&ck entry specified
+
+    # get malware-related pids if requested by configuration
+    malware_pids = None
+    if analyze_malware_pids_only:
+        malware_pids = list(
+            get_malware_processes(
+                metadata_path=analysis_dir / "metadata.json",
+                inject_path=analysis_dir / "inject.log",
+                pstree_path=analysis_dir / "process_tree.json",
+            )
+        )
+
+    # make sure either static or dynamic capability extraction is on
+    assert perform_dynamic_analysis or perform_static_analysis
+
+    # extract capabilities from the Drakvuf report
+    if perform_dynamic_analysis:
+        dynamic_capabilities = dynamic_capa_analysis(
+            analysis_dir, rules, malware_pids=malware_pids
+        )
+
+        # write the extracted TTPs to the analysis dir
+        with (analysis_dir / "ttps.json").open("w", encoding="utf8") as f:
+            for line in construct_ttp_blocks(rules, [dynamic_capabilities], filter_function=lambda rule: rule.meta.get("att&ck", None)):
+                json.dump(line, f)
```

Comment:
Since we already import a "fast JSON" library (msgpack.json, but I suggested orjson above) we can use it here as well.